runtime/vam/agg.go

```diff
@@ -133,18 +176,34 @@ func (c *Sum) Pull(done bool) (vector.Any, error) {
 			c.done = true
 			return c.materialize(c.zctx, c.name), nil
 		}
-		if nulls, ok := vec.(*vector.Nulls); ok {
-			vec = nulls.Values()
+		c.update(vec)
+	}
+}
+
+func (c *Sum) update(vec vector.Any) {
+	if vec, ok := vec.(*vector.Variant); ok {
+		for _, vec := range vec.Values {
+			c.update(vec)
+		}
+		return
+	}
+	vec = c.field.Eval(vec)
+	switch vec := vec.(type) {
+	case *vector.Int:
+		for _, x := range vec.Values {
+			c.sum += x
+		}
+	case *vector.DictInt:
+		for k, val := range vec.Values {
+			c.sum += val * int64(vec.Counts[k])
 		}
-		if vec, ok := vec.(*vector.Int); ok {
-			for _, x := range vec.Values {
-				c.sum += x
-			}
+	case *vector.Uint:
+		for _, x := range vec.Values {
+			c.sum += int64(x)
 		}
-		if vec, ok := vec.(*vector.Uint); ok {
-			for _, x := range vec.Values {
-				c.sum += int64(x)
-			}
+	case *vector.DictUint:
+		for k, val := range vec.Values {
```

Comment:
Nit: Use the same variable names in all of these loops but avoid `val` since that's a `zed.Value` almost everywhere else. `x` is good, though I like `v`, which we often use for this sort of generic Go value.

Reply:
I don't like x.  v is ok.  I think val is ok because when you are in vector world a value vector is called val.  If we want to change this convention there are a lot of places in VNG where we would need to change it.  This code in particular is going away soon.